pkg/timelock/timelock.go

```diff
@@ -239,13 +262,53 @@ func (tw *Worker) subscribeNewLogs(ctx context.Context) (<-chan struct{}, <-chan
 	return done, logCh, nil
 }
 
+// pollNewLogs periodically retrieves logs from the Timelock and emit them through the channel it returns.
+func (tw *Worker) pollNewLogs(ctx context.Context) (<-chan struct{}, <-chan types.Log, error) {
+	lastBlock := tw.fromBlock
+	logCh := make(chan types.Log)
+	done := make(chan struct{})
+
+	go func() {
+		defer close(done)
+		defer close(logCh)
+
+		tw.logger.Debug().Msgf("polling for new logs every %d seconds", tw.listenerPollPeriod)
+		ticker := time.NewTicker(time.Duration(tw.listenerPollPeriod) * time.Second)
+		defer ticker.Stop()
+
+		for {
+			lastBlock = tw.fetchAndDispatchLogs(ctx, logCh, lastBlock)
+
+			select {
+			case <-ticker.C:
+				continue
+			case <-ctx.Done():
+				tw.logger.Debug().Msg("context done; stopping pollNewLogs")
+				SetReadyStatus(HealthStatusError)
+				return
+			}
+		}
+	}()
+
+	return done, logCh, nil
+}
+
 // retrieveHistoricalLogs returns a types.Log channel and retrieves all the historical events of a given contract.
 // Once all the logs have been sent into the channel the function returns and the channel is closed.
 func (tw *Worker) retrieveHistoricalLogs(ctx context.Context) (<-chan struct{}, <-chan types.Log, error) {
-	query := tw.setupFilterQuery()
+	query := tw.setupFilterQuery(tw.fromBlock)
 	logCh := make(chan types.Log)
 	done := make(chan struct{})
 
+	// FIXME(gustavogama-cll): find a more elegant solution to skip retrieving historical
```

Comment:
I left this as is for now to avoid changing the existing implementation too much.

We can incrementally refactor the code later, after the websocket deadline.